README.md

```diff
@@ -1,73 +1,207 @@
+  currentRound: number;
+  players:[{
+    socketId: string;
+    nickName: string;
+    deactivated: string;
+  }];
+  rounds:[
+    roundNumber: number;
+  choices: [
+    userSocket: string;
+  choice: Shapes;
+];
+  winner: string;
+];
+  paused: boolean;
+  finished: boolean;
+  pausedBy: string;
+}
 ```
 
-## Test
+### OutgoingEvents.CONNECTED_SUCCESSFULLY
+Send after a connection was established. Used for saving on client side user's socket ID.
 
-```bash
-# unit tests
-$ npm run test
 
-# e2e tests
-$ npm run test:e2e
+## WS incoming Events
+
+```javascript
+enum IncomingEvents {
+  START_NEW_GAME = 'startNewGame',
+  JOIN_GAME = 'joinGame',
+  MAKE_CHOICE = 'makeChoice',
+  GET_GAMES_LIST = 'getGamesList',
+  MOVE_NEXT_ROUND = 'moveNextRound',
+  PAUSE_GAME = 'pauseGame',
+  RESUME_GAME = 'resumeGame',
+  QUIT_GAME = 'quitGame',
+}
+```
+
+### IncomingEvents.START_NEW_GAME
+
+#### Request:
+```javascript
+interface StartNewGameMessage {
+  nickName: string;
+  socketId: string;
+  roundsCount: number;
+}
+```
+
+#### Responce:
+See: *OutgoingEvents.GAME_RESPONSE*
 
-# test coverage
-$ npm run test:cov
+### IncomingEvents.JOIN_GAME
+
+#### Request:
+```javascript
+interface JoinNewGameMessage {
+  gameId: string;
+  nickName: string;
+  socketId: string;
+}
+```
+
+#### Responce:
+See: *OutgoingEvents.GAME_RESPONSE*
+
+
+### IncomingEvents.QUIT_GAME
+
+#### Request:
+```javascript 
+interface ExitGameMessage {
+    gameId: string;
+    socketId: string;
+}
 ```
 
-## Support
+#### Responce:
+See: *OutgoingEvents.GAME_RESPONSE*
 
-Nest is an MIT-licensed open source project. It can grow thanks to the sponsors and support by the amazing backers. If you'd like to join them, please [read more here](https://docs.nestjs.com/support).
+### IncomingEvents.MAKE_CHOICE
 
-## Stay in touch
+#### Request:
+```javascript
+interface MakeChoiceMessage {
+  gameId: string;
+  socketId: string;
+  choice: Shapes;
+}
+```
+
+#### Responce:
+See: *OutgoingEvents.GAME_RESPONSE*
+
+### IncomingEvents.GET_GAMES_LIST
+
+#### Request:
+none
+
+#### Response:
+Returns array of GameItem
+```javascript
+interface GameItem {
+  id: string;
+  roundsCount: number;
+}
+```
 
-- Author - [Kamil Myśliwiec](https://kamilmysliwiec.com)
-- Website - [https://nestjs.com](https://nestjs.com/)
-- Twitter - [@nestframework](https://twitter.com/nestframework)
+### IncomingEvents.MOVE_NEXT_ROUND
+
+#### Request:
+```javascript
+interface GotToNextRoundMessage {
+  gameId: string;
+  socketId: string;
+}
+```
+
+#### Responce:
+See: *IncomingEvents.GAME_RESPONSE*
+
+### OutgoingEvents.PAUSE_GAME
+
+#### Request:
+```javascript
+interface GotToNextRoundMessage {
+  gameId: string;
+  socketId: string;
+}
+```
+
+#### Responce:
+See: *IncomingEvents.GAME_RESPONSE*
+
+
+### OutgoingEvents.RESUME_GAME
+
+#### Request:
+```javascript
+interface GotToNextRoundMessage {
+  gameId: string;
+  socketId: string;
+}
+```
 
-## License
+#### Responce:
+See: *OutgoingEvents.GAME_RESPONSE*
```

Comment:
As this is an educational project, I would recommend you to add some links in this file, for all the technologies used.

Reply:
Sure.